### ISSUE TITLE
Add Belarusian language support

### ISSUE THREAD

**Issue by timfunky** [sspanak/tt9#675]
Hello @sspanak or should I say zdravey. First of all, thank you a bunch for your immense contribution to the community of dumb phones and open source in general! Blagodarq!

I would like to contribute by adding Belarusian language support. While creating a layout is not a big deal (letters і, ў, ґ should be supported, right?), I wonder what is the way to create a .csv dictionary. I thought of parsing books in Belarusian, maybe dictionaries (the problem is lacking conjugation there, however), and then putting parsed words in the csv file, avoiding repetitions and one letter words. Then generating frequency value for words(I don't really understand that part, how can I calculate it?).

I am probably reinventing the wheel, so I would like to ask if you or someone what tools/databases/other means you use to make dictionaries for languages.

Thank you for reading and have a nice day.

**Comment by sspanak** [comment]
Вітаю, тімфанкі! Character support is not a problem at all. Also, don't worry about the word frequencies, I'll extract them from the general Android dictionary and inject them in the word list. I'll also take care of the rest of the technical details. I have all the necessary tools in the repo.

If you would like to help, please try finding a good Belarusian word list. Extracting from books and subtitles often produces words with spelling mistakes or words with numbers, punctuation or special characters. And, in the case of Slavic languages, a lot of verb conjugations and noun cases will be missing. 

Instead, I recommend looking for a dictionary from a respected academy or university. In many countries, they have "the big dictionary of X". Such dictionaries are great, they are spell checked and verified by linguists. This results in very high suggestions when you type. The problem is I do not speak Belarusian and it is very difficult for me to search in Belarusian websites. On the other hand, your help as a native speaker will be invaluable. This is all I am asking.

Also, I would like to clarify the letter order for ABC mode. Should 3-key be: `дежзё` or `деёжз`? Also, where do "ш", "ы", "ь" and "э" go? In Bulgarian, we have 8: "шщъь", 9: "юя", but I can see the positions of all letter from "ч" onward vary between all Slavic languages. There is even inconsistency between some keyboards and devices when using the same language. I really hope you know what people are used to and what will feel just right.

And after we are done, I hope you will enjoy using TT9 even more!

Btw, you've found the most incorrect way of spelling "Blagodarq". :laughing: It should be either "Blagodarya" or "Благодаря".

**Comment by sspanak** [comment]
@timfunky, were you able find a good dictionary? Also, I still need know what the letter order and locations should be.

**Comment by saw55or-ship-it** [comment]
> @timfunky, were you able find a good dictionary? Also, I still need know what the letter order and locations should be.

I put the letters in the correct order.  I drew the last sketch again so that there were no blots

<img width="792" height="593" alt="Image" src="https://github.com/user-attachments/assets/d440cbbe-9772-419c-a34e-39fe86b073e7" />
<img width="2448" height="3264" alt="Image" src="https://github.com/user-attachments/assets/41e94621-bc05-4c64-8de0-99ba235faa81" />

**Comment by sspanak** [comment]
Awesome, thanks!

Just to confirm, Ц and Ч are on different keys? Are you sure about that? I would expect them to be next to each other on 7 like in all other languages. I was only unsure about "ш", "ы", "ь", "э", because they tend to vary sometimes.

Also, I have found a dictionary based on the "наркамаўка" orthography. Is that what people would expect? If the classical orthography is the better option, I will also need help finding another dictionary.

**Comment by saw55or-ship-it** [comment]
> Awesome, thanks!
> 
> Just to confirm, Ц and Ч are on different keys? Are you sure about that? I would expect them to be next to each other on 7 like in all other languages. I was only unsure about "ш", "ы", "ь", "э", because they tend to vary sometimes.
> 
> Also, I have found a dictionary based on the "наркамаўка" orthography. Is that what people would expect? If the classical orthography is the better option, I will also need help finding another dictionary.

Yes,  наркамау`ка is present orthography of the language. It is used now
Oop yes you're right,  ц-ч are better to stand together. So 7ф х ц ч	 8ш ы ь
